merf/merf_test.py

```diff
@@ -178,86 +169,175 @@ def test_pickle(self):
+
+        # Valid: pandas dataframe with strings as column indices
+        self.assert_valid_fit_input(X)
+        self.assert_valid_fit_input(X, cluster_column="cluster")
+        self.assert_valid_fit_input(X, cluster_column="cluster", random_effects=["Z"])
+        self.assert_valid_fit_input(X, cluster_column="cluster", fixed_effects=["X_0", "X_1", "X_2"])
+        self.assert_valid_fit_input(
+            X, cluster_column="cluster", fixed_effects=["X_0", "X_1", "X_2"], random_effects=["Z"]
+        )
+        self.assert_valid_fit_input(X, fixed_effects=["X_0", "X_1", "X_2"], random_effects=["Z"])
+        self.assert_valid_fit_input(X, fixed_effects=["X_0", "X_1", "X_2"])
+        self.assert_valid_fit_input(X, random_effects=["Z"])
+
+        # Valid: numpy array with ints as column indices
+        self.assert_valid_fit_input(np.array(X))
+        self.assert_valid_fit_input(np.array(X), cluster_column=4)
+        self.assert_valid_fit_input(np.array(X), cluster_column=4, random_effects=[3])
+        self.assert_valid_fit_input(np.array(X), cluster_column=4, fixed_effects=[0, 1, 2])
+        self.assert_valid_fit_input(np.array(X), cluster_column=4, fixed_effects=[0, 1, 2], random_effects=[3])
+        self.assert_valid_fit_input(np.array(X), fixed_effects=[0, 1, 2], random_effects=[3])
+        self.assert_valid_fit_input(np.array(X), fixed_effects=[0, 1, 2])
+        self.assert_valid_fit_input(np.array(X), random_effects=[3])
+
+        # Valid: pandas array with ints as column indices
+        self.assert_valid_fit_input(X)
+        self.assert_valid_fit_input(X, cluster_column=4)
+        self.assert_valid_fit_input(X, cluster_column=4, random_effects=[3])
+        self.assert_valid_fit_input(X, cluster_column=4, fixed_effects=[0, 1, 2])
+        self.assert_valid_fit_input(X, cluster_column=4, fixed_effects=[0, 1, 2], random_effects=[3])
+        self.assert_valid_fit_input(X, fixed_effects=[0, 1, 2], random_effects=[3])
+        self.assert_valid_fit_input(X, fixed_effects=[0, 1, 2])
+        self.assert_valid_fit_input(X, random_effects=[3])
+
+        # Valid: pandas array with mixes type column indices
+        self.assert_valid_fit_input(X)
+        self.assert_valid_fit_input(X, cluster_column=4)
+        self.assert_valid_fit_input(X, cluster_column=4, random_effects=["Z"])
+        self.assert_valid_fit_input(X, cluster_column="cluster", fixed_effects=[0, 1, 2])
+        self.assert_valid_fit_input(X, cluster_column=4, fixed_effects=["X_0", "X_1", "X_2"], random_effects=[3])
+        self.assert_valid_fit_input(X, fixed_effects=[0, 1, 2], random_effects=[3])
+        self.assert_valid_fit_input(X, fixed_effects=["X_0", "X_1", "X_2"])
+        self.assert_valid_fit_input(X, random_effects=["Z"])
+
+        # Valid: Z not as list
+        self.assert_valid_fit_input(X, random_effects="Z")
+        self.assert_valid_fit_input(X, random_effects=3)
+        self.assert_valid_fit_input(np.array(X), random_effects=3)
```

Comment:
You are testing numpy and pandas, but not simpler formats like a nested list.
Could you add test where X is list of lists?

```suggestion
        self.assert_valid_fit_input(np.array(X), random_effects=3)
        self.assert_ialid_fit_input(X.T.values.tolist(), cluster_column=4)
```

Reply:
Done